bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -161,17 +163,39 @@ with h5py.File(args.template_file, "r") as f_src:
 temp_samples = numpy.vstack((template_pars)).T
 
 
+# Rejection Sampling
+if 'mchirp' not in args.fit_param:
+    raise ValueError("mchirp does not exist in args.fit_param")
+mchirp_index = args.fit_param.index('mchirp')
+if args.mchirp_downsample_power is not None:
+    power_vals = template_pars[mchirp_index] ** args.mchirp_downsample_power
+    f_dest.attrs['mchirp_downsample_power'] = args.mchirp_downsample_power
+    probabilities = power_vals / numpy.max(power_vals)
+    rand_nums = numpy.random.uniform(0, 1, len(mass1))
+    ind = rand_nums < probabilities
+    new_mchirp = template_pars[mchirp_index][ind]
+    additional_params = numpy.delete(temp_samples, mchirp_index, axis=1)[ind]
+    new_temp_samples = numpy.column_stack((new_mchirp, additional_params))
+
+
 if args.make_template_kde:
     logging.info('Starting optimization of template KDE parameters')
-    optbw, optalpha = optimizedparam(temp_samples, alphagrid=args.alpha_grid,
+    optbw, optalpha = optimizedparam(new_temp_samples, alphagrid=args.alpha_grid,
                                      bwgrid=args.bw_grid, nfold=args.nfold_template)
     logging.info('Bandwidth %.4f, alpha %.2f' % (optbw, optalpha))
     logging.info('Evaluating template KDE')
-    template_kde = kde_awkde(temp_samples, temp_samples, alp=optalpha,
+    template_kde = kde_awkde(new_temp_samples, temp_samples, alp=optalpha,
                              gl_bandwidth=optbw)
+    
+    compensation_factor = numpy.ones_like(template_kde)
+    if args.mchirp_downsample_power is not None:
+        mchirp_downsample_factor = power_vals[ind]
```

Comment:
why are you indexing on 'ind' here? we evaluate the KDE on all the templates, the KDE itself is a smooth function that needs to be compensated for the downsampling factor everywhere, I don't think it makes sense to do the compensation for only a subset of templates